src/proxy/cache/proxy.js

```diff
@@ -356,18 +356,20 @@ export default _.defaults(
              * @returns {Promise} resolves with the action result
              */
             this.syncData = () => {
+                let actions;
                 return this.queue.serie(() => {
                     return this.actiontStore
                         .flush()
                         .then(data => {
+                            actions = data;
                             if (data && data.length) {
                                 return this.send('sync', data);
                             }
                         })
                         .catch(err => {
                             if (this.isConnectivityError(err)) {
                                 this.setOffline('communicator');
-                                _.forEach(data, action => {
+                                _.forEach(actions, action => {
```

Comment:
The only one that looks like a bug before my fix. I refactored this function to the state it was in before `actions` disappeared.